components/automate-gateway/gateway/services.go

```diff
@@ -662,10 +665,21 @@ func (s *Server) authRequest(r *http.Request,
 	return authzResp.Err()
 }
 
-func logError(err error, resource, action string) {
-	log.WithFields(log.Fields{
-		"error":    err,
-		"resource": resource,
-		"action":   action,
-	}).Error("error authorizing request")
+// XXX needs headers, doesn't it?
+func getProjectsFromMetadata(projectHeaderEntries []string) []string {
+	if projectHeaderEntries == nil {
```

Comment:
👍 but also 😢

Reply:
Sorry that one should have been deleted -- I've done the refactor here, after all. 😄